x-pack/plugins/security_solution/public/common/lib/cell_actions/add_to_timeline.tsx

```diff
@@ -41,23 +44,38 @@ export const getAddToTimelineCellAction = ({
           [timelines]
         );
 
-        const dataProvider: DataProvider[] = useMemo(
-          () =>
+        const dataProvider: DataProvider[] = useMemo(() => {
+          const queryIdPrefix = `${escapeDataProviderId(columnId)}-row-${rowIndex}-col-${columnId}`;
+          return (
             value?.map((x) => ({
               and: [],
               enabled: true,
-              id: `${escapeDataProviderId(columnId)}-row-${rowIndex}-col-${columnId}-val-${x}`,
-              name: x,
               excluded: false,
               kqlQuery: '',
+              id: `${queryIdPrefix}-val-${x}`,
+              name: x,
               queryMatch: {
                 field: columnId,
                 value: x,
                 operator: IS_OPERATOR,
               },
-            })) ?? [],
-          [columnId, rowIndex, value]
-        );
+            })) ?? [
+              {
+                and: [],
+                enabled: true,
+                kqlQuery: '',
+                id: `${queryIdPrefix}`,
+                name: '',
+                excluded: true,
+                queryMatch: {
+                  field: columnId,
+                  value: '',
+                  operator: EXISTS_OPERATOR,
+                },
+              },
+            ]
+          );
```

Comment:
💅 Now that there is an actual case for when the value doesn't exist, it's a bit harder to read and reason about this hook. Wdyt about making it simpler like this:

```
if (value) {
  return value.map(...);
} else {
  return [{...}]
}
```

Reply:
I was on the fence on making this change but skipped it. I 100% agree that it will make it more readable. will make this fix

Reply:
Done.. but a little bit change in the format like below:
```
if(!value){
     return [{...}]
}

return value.map()
```